lib/libvmod_vtc/vmod_vtc.c

```diff
@@ -372,3 +372,38 @@ vmod_proxy_header(VRT_CTX, VCL_ENUM venum, VCL_IP client, VCL_IP server,
 	return (VRT_blob(ctx, "proxy_header", h, l,
 	    BLOB_VMOD_PROXY_HEADER_TYPE));
 }
+
+/*--------------------------------------------------------------------*/
+
+VCL_DURATION v_matchproto_(td_vtc_benchmark)
+vmod_benchmark(VRT_CTX, VCL_SUB sub, VCL_INT total, VCL_INT leg)
+{
+	VCL_HTTP where;
+
+	VCL_INT i, count = 0;
+
+	vtim_mono t_tot = 0;
+	vtim_mono b, e;
+
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
+
+	where = ctx->http_req ? ctx->http_req : ctx->http_bereq;
+
+	while (count < total) {
+		if (WS_Overflowed(ctx->ws))
+			break;
```

Comment:
Shouldn't the method fail here, or at least somehow make it recognizable that not all of the requested iterations were executed? Suppose that workspace overflows after very few iterations, even only one, or none at all. The the benchmark result does not have the benefit of averaging, and could be very non-representative. How could you know that happened?

Maybe write a message to the log here?
Maybe a summary for the log, including the number of iterations, in the successful case (no overflows)?

Reply:
true, thank you. i have added a `VRT_fail()`